chapter2.adoc

```diff
@@ -134,68 +132,147 @@ The privilege level of the trigger chain is determined by the trigger enabled fo
-| Field | Access condition
-| m     | mdbgen == 1                           
-| s     | mdbgen == 1 \|\| sdedbgalw == 1    
-| u     | mdbgen == 1 \|\| sdedbgalw == 1   
-| vs    | mdbgen == 1 \|\| sdedbgalw == 1  
-| vu    | mdbgen == 1 \|\| sdedbgalw == 1
-|====================================
-
-=== Other CSR updates
+[cols="25%,25%,50%"]
+|============================================================================================
+| Number   | Name       | Descirption
+| 0xaaa    | sdcsr      | Supervisor debug control and status register.
+| 0xaaa    | sdpc       | Supervisor debug program counter. 
+| 0xaaa    | sdcratch0  | Supervisor debug scratch register 0.
+| 0xaaa    | sdcratch1  | Supervisor debug scratch register 1.
+|============================================================================================
 
-==== Debug Control and Status (dcsr)
+The `sdcsr` provides supervisor access to `dcsr` and is a subset of it. The accessible fields are formatted as shown in <<sdcsr32>> and <<sdcsr64>>. When `mdbgen` is 0 and `sdedbgalw` is set to 1, the `prv` and `v` fields indicate the the privilege level at which the hart was executing before entering Debug Mode. Meanwhile, the `sdpc` is updated with the address of the next instruction to executed upon entry into Debug Mode. When resuming from Debug Mode, the privilege level of the hart is restored to the values in `prv` and `v`, while the hart's PC is updated wth the address in `sdpc`.   
 
-The dcsr is always accessible in Debug Mode. The access rules for field `prv` and `v` are addressed in subsection <<prvvacc>>. Beside `prv` and `v`, the access condition of remaining fields are listed in the following table.
-When the access conditions are met, they are read/write accessible. When access is disallowed, writes to these fields are ignored, and reads return zero.
+The `sdscratch0` and `sdscratch1` are optional scratch registers are only required if dscratch0 and dscratch1 are implemented.
 
-.Dcsr fields access condition against privilege granted to external debugger
-[options="header"]
-[cols="40%,60%"]
-|============================================
-| Field    |  Access condition
-| debugver |  mdbgen == 1 \|\| sdedbgalw == 1
-| extcause |  mdbgen == 1 \|\| sdedbgalw == 1
-| cetrig   |  mdbgen == 1 
-| ebreakvs |  mdbgen == 1 \|\| sdedbgalw == 1
-| ebreakvu |  mdbgen == 1 \|\| sdedbgalw == 1
-| ebreakm  |  mdbgen == 1
-| ebreaks  |  mdbgen == 1 \|\| sdedbgalw == 1
-| ebreaku  |  mdbgen == 1 \|\| sdedbgalw == 1
-| stepie   |  mdbgen == 1 \|\| sdedbgalw == 1
-| stoptime |  mdbgen == 1
-| mprven   |  mdbgen == 1
-| nmip     |  mdbgen == 1
-|============================================
-
-==== Debug PC (dpc) and Debug Scratch Register (dscratch0 and dscratch1)
+[NOTE]
+In a straightforward implementation, reading or writing any field in `sdcsr` is equivalent to reading or writing the corresponding field in `dcsr`, much like how it works for `sstatus`.
```

Comment:
I don't think there's any alternate implementation option here.  I think the first paragraph in this section makes this clear, and this note should probably be removed.

Reply:
Right, thanks for reminding.  I will remove it.